modeling/datasets_sc.py

```diff
@@ -0,0 +1,369 @@
+
+        return index, x1, seg_y
+
+    def test(self, model):
+        """Implements the test phase via animaSegPerfAnalyzer"""
+        assert model is not None
+
+        # Convert to list in case we need to assign new values in the next code block
+        adjusted_center_crop_size = list(self.center_crop_size)
+
+        # Center-crop size vs. subvolume size check to see if we need to pad the inputs or not
+        for i in range(3):
+            ccs_i, svs_i = adjusted_center_crop_size[i], self.subvolume_size[i]
+            if ccs_i % svs_i != 0:
+                # Minimally pad the initial center-crop size s.t. it becomes divisible by SV
+                adjusted_center_crop_size[i] = ((ccs_i // svs_i) * svs_i) + svs_i
+
+        # Convert back to tuple for the center-crop parameter to be immutable again
+        adjusted_center_crop_size = tuple(adjusted_center_crop_size)
+
+        # Report center-crop size parameters
+        print('Original Center-Crop Size: ', self.center_crop_size)
+        if adjusted_center_crop_size != self.center_crop_size:
+            print('[WARNING] Adjusted Center-Crop Size: ', adjusted_center_crop_size)
+
+        for subject_no, subject in enumerate(tqdm(self.subjects_hold_out, desc='Loading Volumes')):
+
+            # Another for loop for going through sessions
+            temp_subject_path = os.path.join(self.root, subject)
+            num_sessions_per_subject = sum(os.path.isdir(os.path.join(temp_subject_path, pth)) for pth in os.listdir(temp_subject_path))
+
+            for ses_idx in range(1, num_sessions_per_subject+1):
+                # Get paths with session numbers
+                session = 'ses-0' + str(ses_idx)
+                subject_images_path = os.path.join(self.root, subject, session, 'anat')
+                subject_labels_path = os.path.join(self.root, 'derivatives', 'labels', subject, session, 'anat')
+
+                # Read original subject image (i.e. 3D volume) to be used for training
+                sag_img = nib.load(os.path.join(subject_images_path, '%s_%s_acq-sag_T2w.nii.gz' % (subject, session)))
+                # hfac, wfac, dfac = sag_img.header.get_zooms()
+                # Read original subject ground-truths (GT)
+                gt = nib.load(os.path.join(subject_labels_path, '%s_%s_acq-sag_T2w_lesion-manual.nii.gz' % (subject, session)))
+
+                # Check if image sizes and resolutions match
+                assert sag_img.shape == gt.shape
+
+                # Convert to Numpy
+                sag_img, gt = sag_img.get_fdata(), gt.get_fdata()
+
+                # Apply center-cropping
+                center_crop = CenterCrop(size=self.center_crop_size)
+                sag_img = center_crop(sample=sag_img, metadata={'crop_params': {}})[0]
+                gt = center_crop(sample=gt, metadata={'crop_params': {}})[0]
+
+                if self.center_crop_size == self.subvolume_size:    # Move on to the prediction
```

Comment:
Inside this `if` statement, all references to `sag_img_subvolume` should be renamed as `sag_img` as `sag_img_subvolume` is only available for the `else` statement.

Reply:
Yeah, this is a good point, I was too lazy to modify everything, so I left it as it was. Will update them now!